### ISSUE TITLE
In documentation please include header file 

### ISSUE THREAD

**Issue by lfmunoz** [catchorg/Catch2#2920]
Title explains issue, but for example:

https://github.com/catchorg/Catch2/blob/devel/docs/generators.md

Copy paste trying learn how it works, but unfortunately I have to figure out which file to include. 

Apparently it is `#include <catch2/generators/catch_generators.hpp>`

Makes it hard to follow or quickly build something, would be very helpful if you link to hpp file needed for the examples shown.

**Comment by ChrisThrasher** [comment]
See https://github.com/catchorg/Catch2/pull/2958
